sdk_contrib/typed_span/src/main/java/io/opentelemetry/sdk/contrib/typedspan/BaseSpanWrapper.java

```diff
@@ -0,0 +1,15 @@
+package io.opentelemetry.sdk.contrib.typedspan;
+
+import io.opentelemetry.trace.Span;
+
+public abstract class BaseSpanWrapper {
+  Span span;
```

Comment:
It doesn't feel like this is pulling much weight, and maybe just adding extra mental overhead. Do you imagine more things going in here?

Reply:
We might insert some hook mechanism for the auto-instrumentation people.

Reply:
After seeing open-telemetry/opentelemetry-auto-instr-java#195, I think what the auto-instrumentation SIG calls typed spans is actually unrelated to this PR.